src/macros.rs

```diff
@@ -42,3 +42,22 @@ macro_rules! interact_or {
         }
     };
 }
+
+#[macro_export]
+macro_rules! check_update {
```

Comment:
It's a macro because I can overload the call signature of it to default the second parameter to `false`, but I can make it into a function if you want.

Reply:
I think a function is preferred here for simplicity. If you want to avoid having to provided `false` as the second parameter, we can have a `check_update_command` function that passed false

Reply:
That sounds good to me. I just finished finals, so I'll finalize this now.